packages/starknet-snap/src/rpcs/executeTxn.test.ts

```diff
@@ -0,0 +1,190 @@
+import type { Invocations, UniversalDetails } from 'starknet';
+import { TransactionType, constants } from 'starknet';
+
+import invocationExamples from '../__tests__/fixture/invocationExamples.json'; // Assuming you have a similar fixture
+import { STARKNET_SEPOLIA_TESTNET_NETWORK } from '../utils/constants';
+import * as starknetUtils from '../utils/starknetUtils';
+import { executeTxn as executeTxnUtil } from '../utils/starknetUtils';
+import {
+  mockAccount,
+  prepareConfirmDialog,
+  prepareMockAccount,
+} from './__tests__/helper';
+import type { ExecuteTxnParams } from './executeTxn';
+import { executeTxn } from './executeTxn';
+
+const prepareMockExecuteTxn = (
+  transactionHash: string,
+  accountDeployed: boolean,
+) => {
+  const executeTxnRespMock = {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    transaction_hash: transactionHash,
+  };
+
+  const getEstmatedFeesRepsMock = {
+    suggestedMaxFee: BigInt(1000000000000000).toString(10),
+    overallFee: BigInt(1000000000000000).toString(10),
+    includeDeploy: !accountDeployed,
+    unit: 'wei',
```

Comment:
with hardcoded unit, it seem it will missing a test case for txn v3?

Reply:
Let's change this in next pr, wdyt ?